nim-mode.el

```diff
@@ -233,9 +233,10 @@ instead.  The default regex’s matching word is [Package]."
                    (conf-mode)
                  (nimscript-mode))))))))
 
-;;; `auto-mode-alist'
-;;;###autoload (add-to-list 'auto-mode-alist '("\\.nim\\'" . nim-mode))
-;;;###autoload (add-to-list 'auto-mode-alist '("\\.nim\\(ble\\|s\\)\\'" . nimscript-mode-maybe))
+;;;###autoload
```

Comment:
I'd prefer shorter version if it's doing same. Is there reason doing this?

Reply:
for some unknown reason to me, the auto-mode-alist wasn't updated when I loaded the module from a custom path. Changing this to two separate lines did fix it. I can test if it works now again.

Reply:
I did some investigations now and figured out what autoload is actually doing. Autoload is a mockup generator to delay the loading of the library. So instead of loading the library a mock library is loaded on startup, and when the functionality is actually invoked the real library gets loaded and replaces the mock library.  The comments are tags to tag functions that should be part of this mock interface. The autoload commend with a command like above puts functions in the autoload mock. But in comment form those commands are only executed, when the actual mock is loaded, not when the real library is loaded. I loaded the library with ``(load-library "nim-mode")`` and circumvented the whole autoload indirection. But here the ``auto-mode-alist`` was never updated.  With the newline there, the `add-to-list` command will be executed twice, when the library is loaded, and when the mock library is loaded. I don't think that is a problem since load-library prevents duplicates, but worth mentioning. I think `(load-library "nim-mode")` is something that should work out of the box.

Reply:
To be honest, I don't think I need to support load-library users in this case. They can make autoload file by `update-directory-autoloads`. I think you are saying for your convenience, but if you load files different order from other users who install a package manager like MELPA, I would say it's out of support. Besides, if you wanted to be a maintainer, how come to use different loading order from normal users?

Reply:
You can simply use el-get (https://github.com/dimitri/el-get) if you don't want learn internal.

Reply:
My first thoughts about ``el-get`` are: "ugh... yet another dependency". So my feelings towards it are already negative. So I have two simple questions regarding it, why is ``el-get`` necessary to be able to work on a fork of nim-mode? And what is the line I have to my init.el to load my fork of nim-mode?  I couldn't figure that out from reading the readme of ``el-get``.

Reply:
but ``el-get`` really shouldn't be discussed here. I reverted the changes regarding autoload. My question is, can this PR be merged now?